code/modules/power/singularity/singularity.dm

```diff
@@ -446,3 +447,37 @@
 	explosion(src.loc,(dist),(dist*2),(dist*4))
 	qdel(src)
 	return(gain)
+
+/obj/singularity/vomit
+	name = "vomit-charged singularity"
+	desc = "It's a gravitational singularity made out of vomit. There appears to be a giant goose stuck inside."
+	icon = 'icons/obj/gooseularity.dmi'
+	icon_state = "gooseularity"
+	pixel_x = -96
+	pixel_y = -96
+	density = FALSE
+	current_size = 9 //It moves/eats like a max-size singulo, aside from range.
+	contained = 0 //Are we going to move around?
+	dissipate = 0 //Do we lose energy over time?
+	move_self = 1 //Do we move on our own?
+	grav_pull = 0 //How many tiles out do we pull?
+	consume_range = 5 //How many tiles out do we eat
+	CanSupermatter = FALSE
+	light_power = 0.7
+	light_range = 7
+	light_color = rgb(0, 255, 0)
+
+/obj/singularity/vomit/consume(atom/A)
+	if (istype(A, /obj/effect/decal/cleanable/vomit))
+		return
+	var/turf/Tgoose = get_turf(src)
+	var/turf/spawnloc = isturf(A) ? A : get_turf(A)
+	playsound(Tgoose, 'sound/effects/splat.ogg', 50, 1)
```

Comment:
``playsound(src, ...`` and remove line 473.
Also the fourth argument is a boolean.

Reply:
i have to get the turf or else it won't play any sound

Reply:
wrong. https://github.com/tgstation/tgstation/blob/master/code/game/sound.dm